src/components/side_panel/find_and_replace/find_and_replace_store.ts

```diff
@@ -307,18 +312,58 @@ export class FindAndReplaceStore extends SpreadsheetStore implements HighlightPr
       matches: [this.searchMatches[this.selectedMatchIndex]],
       searchOptions: this.searchOptions,
     });
+
+    if (this.irreplaceableMatchCount > 0) {
+      this.showReplaceWarningMessage(1, this.irreplaceableMatchCount);
+    }
+    this.irreplaceableMatchCount = 0;
+
     this.selectNextCell(Direction.next);
   }
   /**
    * Apply the replace function to all the matches one time.
    */
   replaceAll() {
+    const searchMatches = this.searchMatches;
+
     this.model.dispatch("REPLACE_SEARCH", {
       searchString: this.toSearch,
       replaceWith: this.toReplace,
-      matches: this.searchMatches,
+      matches: searchMatches,
       searchOptions: this.searchOptions,
     });
+
+    if (this.irreplaceableMatchCount > 0) {
+      this.showReplaceWarningMessage(searchMatches.length, this.irreplaceableMatchCount);
+    }
+    this.irreplaceableMatchCount = 0;
```

Comment:
this little duplicated snippet of code could be moved at `case "REPLACE_SEARCH"`, right ?

Reply:
Yes, good idea! 🔄

Changes made, Thanks!